osquery/events/events.cpp

```diff
@@ -139,8 +139,8 @@ std::vector<std::string> EventSubscriberPlugin::getIndexes(EventTime start,
       auto step = boost::lexical_cast<EventTime>(bin);
       // Check if size * step -> size * (step + 1) is within a range.
       int bin_start = size * step, bin_stop = size * (step + 1);
-      if (expire_events_ && step * size < expire_time_) {
-        expirations.push_back(list_type + "." + bin);
+      if (expire_events_ && bin_stop < expire_time_) {
+        expirations.push_back(bin);
```

Comment:
This seems to address an events expiration problem, but expirations still occur at the list binning level. To log a continuous stream of events we'll need to apply eviction at the second level (most precise level of event ordering) OR apply logic at the logger plugin. Currently, the best way to log an event stream is by setting: `"removed": false` in the scheduled query.